bootstrap/Dockerfile

```diff
@@ -31,22 +20,38 @@ RUN wget -q https://dl.google.com/dl/cloudsdk/channels/rapid/google-cloud-sdk.ta
 RUN ln -sf /google-cloud-sdk/bin/gcloud /usr/local/bin/gcloud
 
 # Install kubectl
-# We don't install via gcloud because we want 1.10 which is newer than what's in gcloud.
-RUN  curl -LO https://storage.googleapis.com/kubernetes-release/release/v1.10.0/bin/linux/amd64/kubectl && \
+# We don't install via gcloud because we want 1.12.2 which is newer than what's in gcloud.
+RUN  curl -LO https://storage.googleapis.com/kubernetes-release/release/v1.12.2/bin/linux/amd64/kubectl && \
     mv kubectl /usr/local/bin && \
     chmod a+x /usr/local/bin/kubectl
 
+# bootstrap is using ksonnet 0.13.1
+RUN wget -q https://github.com/ksonnet/ksonnet/releases/download/v0.13.1/ks_0.13.1_linux_amd64.tar.gz && \
+    tar xzf ks_0.13.1_linux_amd64.tar.gz -C . && \
+    rm ks_0.13.1_linux_amd64.tar.gz && \
+    cp ks_0.13.1_linux_amd64/ks /usr/local/bin && \
+    rm -rf ks_0.13.1_linux_amd64
+
+RUN mkdir -p /opt/kubeflow
+RUN mkdir -p $GOPATH/src/github.com/kubeflow/kubeflow/bootstrap
+WORKDIR $GOPATH/src/github.com/kubeflow/kubeflow/bootstrap
+
+ENV PATH /go/bin:/usr/local/go/bin:$PATH
+# use go modules
+ENV GO111MODULE=on
+
+COPY go.mod .
+COPY go.sum .
+RUN go mod download
+
+FROM bootstrap_build as bootstrap
```

Comment:
Can we use golang:alpine here? and copy from bootstrap_build so we can have smaller output image?

Reply:
@kunmingg yes, great suggestion. 

I'll get this in tomorrow.